base/setup/usetup/mui.c

```diff
@@ -303,6 +303,138 @@ MUIGetEntry(
     return NULL;
 }
 
+/**
+ * @MUIGetEntryTextFromCoord
+ *
+ * Retrieves the portion of text from a MUI entry given the X and Y coordinate points.
+ *
+ * @param[in]   Page
+ *     The MUI (Multilingual User Interface) entry page number, as a unsigned long integer.
+ *
+ * @param[in]   X
+ *      The X coordinate where the text belongs.
+ *
+ * @param[in]   Y
+ *      The Y coordinate where the text belongs.
+ *
+ * @return
+ *     Returns a pointer to a constant string pointed by the MUI entry.
+ *
+ */
+LPCSTR
+MUIGetEntryTextFromCoord(
+    IN ULONG Page,
+    IN SHORT X,
+    IN SHORT Y)
+{
+    const MUI_ENTRY * entry;
+    ULONG index;
+
+    /* Retrieve the entries of a MUI page */
+    entry = FindMUIEntriesOfPage(Page);
+    if (!entry)
+    {
+        DPRINT("MUIGetEntryTextFromCoord(): Failed to get the translated entry page!\n");
+        return NULL;
+    }
+
+    /* Loop over the entry and monitor the coordinate points that match */
+    for (index = 0; entry[index].Buffer != NULL; index++)
+    {
+        if (entry[index].X == X && entry[index].Y == Y)
+        {
+            /* They do so return the text buffer */
+            return entry[index].Buffer;
+        }
+    }
+
+    DPRINT("MUIGetEntryTextFromCoord(): The coordinate areas don't match with those from the retrieved MUI entry!\n");
+    return NULL;
+}
+
+/**
+ * @MUIHighlightText
+ *
+ * The function fills the background of the text in white (highlight).
+ *
+ * @param[in]   Page
+ *     The MUI (Multilingual User Interface) entry page number, as a unsigned long integer.
+ *
+ * @param[in]   X
+ *      The X coordinate where the text should be highlighted.
+ *
+ * @param[in]   Y
+ *      The Y coordinate where the text should be highlighted.
+ *
+ * @param[in]   Row
+ *      Number of rows to be highlighted.
+ *
+ * @return
+ *     Nothing.
+ *
+ */
+VOID
+MUIHighlightText(
+    IN ULONG Page,
+    IN SHORT X,
+    IN SHORT Y,
+    IN SHORT Row)
+{
+    LPCSTR Buffer;
+    ULONG Length;
+
+    Buffer = MUIGetEntryTextFromCoord(Page, X, Y);
+    if (!Buffer)
+    {
+        DPRINT("MUIHighlightText(): Couldn't obtain a buffer from the string text!\n");
```

Comment:
Why this DPRINT since MUIGetEntryTextFromCoord already DPRINTs all failure conditions ?
Why don't we "return" here as using NULL buffer on next seps is likely not a good idea ? Or use a "default" value for Length ?

Reply:
It's my own way to add debug print functions to log debug information as much as possible, verbose-wise. Although I do agree that this function should bail out if no buffer could be captured.

Reply:
ok